### ISSUE TITLE
[BUG]: Breaking change in 4.13.0 - string properties no longer allow null by default

### ISSUE THREAD

**Issue by waf** [aloneguid/parquet-dotnet#360]
### Library Version

4.13.0

### OS

Windows (but I suspect all OSs)

### OS Architecture

64 bit

### How to reproduce?

Run the following code (it's a valid top-level statement program):

```csharp
using Parquet.Serialization;

var data = new[] { new MyPoco { Name = "Joe" } };
var stream = new MemoryStream();
await ParquetSerializer.SerializeAsync(data, stream);

public class MyPoco
{
    public string Name { get; set; }
    public string Location { get; set; }
}
```

The above program works on versions of parquet-dotnet `4.12.0` and below. It started failing on `4.13.0` and above with:

```
System.ApplicationException: 'failed to serialise data column 'Location''
   at Parquet.Serialization.ParquetSerializer.<SerializeRowGroupAsync>d__2`1.MoveNext()
   at Parquet.Serialization.ParquetSerializer.<SerializeAsync>d__3`1.MoveNext()
   at Program.<<Main>$>d__0.MoveNext() in C:\Projects\Personal\ParquetStringTest\ParquetStringTest\Program.cs:line 6

Inner Exception:

Object reference not set to an instance of an object.
   at Parquet.Encodings.ParquetPlainEncoder.Encode(ReadOnlySpan`1 data, Stream destination)
   at Parquet.Encodings.ParquetPlainEncoder.Encode(Array data, Int32 offset, Int32 count, SchemaElement tse, Stream destination, DataColumnStatistics stats)
   at Parquet.File.DataColumnWriter.<WriteColumnAsync>d__12.MoveNext()
   at Parquet.File.DataColumnWriter.<WriteAsync>d__9.MoveNext()
   at Parquet.ParquetRowGroupWriter.<WriteColumnAsync>d__15.MoveNext()
   at Parquet.Serialization.ParquetSerializer.<SerializeRowGroupAsync>d__2`1.MoveNext()
```

I suppose this was broken in https://github.com/aloneguid/parquet-dotnet/pull/341 . 

This behavior contradicts the [documentation on Null Values](https://github.com/aloneguid/parquet-dotnet/blob/035a4910338b41f34f643c383d01042a298806b5/docs/schema.md#null-values):
> Note that string type is always nullable. Although string is an [immutable type](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/strings/) in CLR, it's still passed by reference, which can be assigned to null value.

I believe it's an inadvertent breaking change than an intended change.

**Comment by aloneguid** [comment]
Thanks, looks like a regression indeed.

**Comment by aloneguid** [comment]
@waf shoudl be fixed in 4.15.

**Comment by waf** [comment]
Wow, that was fast. Confirmed as fixed. Thank you!
